ssl/ssl_test.go

```diff
@@ -0,0 +1,581 @@
+package ssl
+
+import (
+	"errors"
+	"strconv"
+	"testing"
+
+	"github.com/SUSE/scf-secret-generator/model"
+	"github.com/cloudflare/cfssl/csr"
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/api/core/v1"
+)
+
+const CERT_ID = "cert-id"
+
+type MockBase struct {
+	mock.Mock
+}
+
+type MockLog struct {
+	MockBase
+}
+
+func (m *MockLog) Fatalf(str string, message ...interface{}) {
+	m.Called(str, message)
+}
+
+type MockSSL struct {
+	MockBase
+}
+
+func (m *MockSSL) createCA(secrets *v1.Secret, updates *v1.Secret, id string) bool {
+	args := m.Called(secrets, updates, id)
+	return args.Bool(0)
```

Comment:
Oh, this is something with the mock library that I don't like or maybe don't understand correctly.

The `Call` object has two members:
* `ReturnArguments` 
* `Arguments`

They are both type `Arguments`, but one is the arguments *to* the function, the other comes from `.On(...).Return(...)`

https://godoc.org/github.com/stretchr/testify/mock#Call
https://godoc.org/github.com/stretchr/testify/mock#Mock.Called

IMO  `ReturnArguments` shouldn't be an `Arguments` type, because it's confusing. I would expect it to be called `ReturnValues` or something like that.

Reply:
Clarified variable names in 3fbac70 and 0225aab